tasks/notifications.yml

```diff
@@ -1,23 +1,39 @@
 ---
-- name: Check alert notifications list
-  uri:
-    url: "{{ grafana_api_url }}/api/alert-notifications"
-    user: "{{ grafana_security.admin_user }}"
-    password: "{{ grafana_security.admin_password }}"
-    force_basic_auth: true
-    return_content: true
-  no_log: true
-  register: alert_notifications
+- block:
+  - name: Check alert notifications list
+    uri:
+      url: "{{ grafana_api_url }}/api/alert-notifications"
+      user: "{{ grafana_security.admin_user }}"
+      password: "{{ grafana_security.admin_password }}"
+      force_basic_auth: true
+      return_content: true
+    no_log: true
+    register: alert_notifications
 
-- name: Create grafana alert notification channels
-  uri:
-    url: "{{ grafana_api_url }}/api/alert-notifications"
-    user: "{{ grafana_security.admin_user }}"
-    password: "{{ grafana_security.admin_password }}"
-    force_basic_auth: true
-    method: POST
-    body_format: json
-    body: "{{ item | to_json }}"
-  with_items: "{{ grafana_alert_notifications }}"
-  no_log: true
-  when: ((alert_notifications['json'] | selectattr("name", "equalto", item['name'])) | list) | length == 0
+  - name: Create grafana alert notification channels
+    uri:
+      url: "{{ grafana_api_url }}/api/alert-notifications"
+      user: "{{ grafana_security.admin_user }}"
+      password: "{{ grafana_security.admin_password }}"
+      force_basic_auth: true
+      method: POST
+      body_format: json
+      body: "{{ item | to_json }}"
+    with_items: "{{ grafana_alert_notifications }}"
+    no_log: true
+    when: ((alert_notifications['json'] | selectattr("name", "equalto", item['name'])) | list) | length == 0
+
+  when: not grafana_use_provisioning
```

Comment:
This can be removed. Idea is to remove all `uri` usage and move to grafana ansible modules. Adding this code moves us back and adds tech-debt.

After removing this, we need an information in README.md stating that `grafana_alert_notifications` can be used only when `grafana_use_provisioning` is enabled.

Reply:
I removed those two tasks and updated README. Could you please take another look?